genestack_client/study.py

```diff
@@ -0,0 +1,88 @@
+# coding=utf-8
+from genestack_client import Application
+
+
+class StudyDesign(Application):
+    """
+    API for working with studies and administrating the Study Design application.
+
+    Study Design is an application for describing studies and samples using Genestack metainfo
+    mechanism.
+    """
+    APPLICATION_ID = 'genestack/study-design'
+
+    def get_study_info(self, study_accession):
+        """
+        Get information about the given study from the project management point of view.
+
+        :param study_accession: study accession
+        :type study_accession: str
+        :return: dictionary with the following entries:
+
+            * ``studyNumber``: number of study unique for the organization,
+            * ``studyStatus``: current workflow step,
+            * ``approved``: mark of passed curator's review,
+            * ``hasApprovalButton``: mark of *Approve/Disapprove* button visibility in UI,
+            * ``approvalButtonHidden``: *true* if *Approve/Disapprove* button was hidden via
+              the :meth:`set_approval_button_visibility` method, *false* otherwise,
+            * ``linkedDataset``: accession of the most recently created dataset with expression data
+              that is linked to this study and can be opened in the Expression Repository
+              application or *None* if no datasets can be found.
+
+        :rtype: dict[str, str]
+        """
+        return self.invoke('getStudyInfo', study_accession)
+
+    def set_study_status(self, study_accession, new_status):
+        """
+        Update status of the given study.
+
+        Status is a name of a project workflow stage as it is going to be displayed in the Study
+        Design UI, e.g. "*Draft*", "*Waiting for review*", etc.
+
+        :param study_accession: study accession
+        :type study_accession: str
+        :param new_status: status name
+        :type new_status: str
+        """
+        self.invoke('setStudyStatus', study_accession, new_status)
```

Comment:
How about some constants and validation?

Reply:
We can use https://pypi.org/project/enum34/  library  (we already have it on machine, because it is request dependency).   In that case we can expect enum instance and don't have additional validation.

Reply:
All logic is contained within the Java code. Status constants are not applicable here, because it can be any string you want